GVFS/GVFS.Common/GVFSEnlistmentStatistics.cs

```diff
@@ -0,0 +1,189 @@
+                });
+
+            return directoriesSortedByHydration;
+        }
+
+        public double GetHydrationOfDirectory(string directory)
+        {
+            if (!this.hydratedFilesDirectoryTracking.ContainsKey(directory))
+            {
+                return 0;
+            }
+
+            return (double)this.hydratedFilesDirectoryTracking[directory] / this.trackedFilesDirectoryTracking[directory];
+        }
+
+        public int GetTrackedFilesCount()
+        {
+            return this.trackedFilesCount;
+        }
+
+        public int GetPlaceholderCount()
+        {
+            return this.placeholderCount;
+        }
+
+        public int GetModifiedPathsCount()
+        {
+            return this.modifiedPathsCount;
+        }
+
+        public double GetPlaceholderPercentage()
+        {
+            if (this.trackedFilesCount == 0)
+            {
+                return 0;
+            }
+
+            return (double)this.placeholderCount / this.trackedFilesCount;
+        }
+
+        public double GetModifiedPathsPercentage()
+        {
+            if (this.trackedFilesCount == 0)
+            {
+                return 0;
+            }
+
+            return (double)this.modifiedPathsCount / this.trackedFilesCount;
+        }
+
+        /// <summary>
+        /// Take a string representing a file path on system and pull out the upper-most directory from it as a string, or '/' if it is in the root
+        /// </summary>
+        /// <param name="path">The path to a file to parse for the top level directory containing it</param>
+        /// <returns>A string containing the top level directory from the provided path, or '/' if the path is for an item in the root</returns>
+        private string ParseTopDirectory(string path)
+        {
+            int whackLocation = path.IndexOf('/');
+            if (whackLocation == -1)
+            {
+                return "/";
+            }
+
+            return path.Substring(0, whackLocation);
+        }
+
+        /// <summary>
+        /// Categorizes a list of paths given as strings by mapping them to the top level directory in their path
+        /// Modifies the directoryTracking dictionary to have an accurate count of the files underneath a top level directory
+        /// </summary>
+        /// <remarks>
+        /// The distinction between files and directories is important --
+        /// If the path to a file doesn't contain a '/', then that means it falls within the root
+        /// However if a directory's path doesn't contain a '/', it counts towards itself for hydration
+        /// </remarks>
+        /// <param name="paths">An enumerable containing paths as strings</param>
+        /// <param name="directoryTracking">A dictionary used to track the number of files per top level directory</param>

```

Comment:
Also add a comment describing `isFile`

Reply:
Added one for both new parameters, good catch!